c7n/resources/fsx.py

```diff
@@ -325,17 +325,32 @@ class DeleteFileSystem(BaseAction):
     dependencies necessary to delete the file system.
 
     You can override the default retry settings for deletion by specifying
-    `retry-delay` (default: 5 seconds) and `retry-max-attempts` (default: 2).
-    FSx for Ontap may take longer to delete all volumes before it can delete
-    the file system.
+    `retry-delay` (default: 1 seconds, if force is True defaults to 30 seconds)
+    and `retry-max-attempts` (default: 1, if force is True defaults to 10).
+    Adjust the retry settings, as necessary when using `force` set to `True`.
+    FSx for Ontap takes extra time to delete all volumes before it can delete
+    the file system. OpenZFS also takes extra time to delete S3 access points.
 
     Note:
-    If `skip-snapshot` is set to True, no final snapshot will be created.
-    FSx for OnTap resources do not create snapshot backups on deletion
-    even if skip-snapshot is set to False.
-    FSx for Lustre resources using the Scratch deployment type do not support
-    final backups on deletion. Set `force` to True to delete these when
-    `skip-snapshot` is set to False.
+
+    - If `skip-snapshot` is set to True, no final snapshot will be created.
+    - FSx for OnTap resources do not create snapshot backups on deletion even \
+      if skip-snapshot is set to False.
+    - FSx for Lustre resources using the Scratch deployment type do not support \
+      final backups on deletion. Set `force` to True to delete these when \
+      `skip-snapshot` is set to False.
+
+    Annotated Permissions:
+
+    - fsx:DeleteFileSystem (required)
+    - fsx:CreateBackup (if skip-snapshot is False or not set)
+    - fsx:DescribeStorageVirtualMachines (if force is True for ONTAP)
+    - fsx:DeleteStorageVirtualMachine (if force is True for ONTAP)
+    - fsx:DescribeVolumes (if force is True for ONTAP and OpenZFS)
+    - fsx:DeleteVolume (if force is True for ONTAP and OpenZFS)
+    - fsx:DescribeS3AccessPointAttachments (if force is True for OpenZFS)
+    - fsx:DetachAndDeleteS3AccessPoint (if force is True for OpenZFS)
+    - s3:DeleteAccessPoint (if force is True for OpenZFS)
```

Comment:
:trophy: thank you for adding the permission detail here, given the complex set of possible permissions this is useful stuff